src/main/scala/com/microsoft/ml/spark/lightgbm/LightGBMRanker.scala

```diff
@@ -61,7 +62,8 @@ class LightGBMRanker(override val uid: String)
   }
 
   def getModel(trainParams: TrainParams, lightGBMBooster: LightGBMBooster): LightGBMRankerModel = {
-    new LightGBMRankerModel(uid, lightGBMBooster, getLabelCol, getFeaturesCol, getPredictionCol)
+    new LightGBMRankerModel(uid, lightGBMBooster, getLabelCol, getFeaturesCol, getPredictionCol,
```

Comment:
Can you make it so that you set the params with setLabelCol as opposed to using a big constructor? I think the constructors are causing a lot of clutter

Reply:
done, changed models to use ComplexParamsWritable based on discussion